llvm/lib/Target/NVPTX/NVPTXAsmPrinter.cpp

```diff
@@ -839,16 +827,14 @@ void NVPTXAsmPrinter::emitGlobalAlias(const Module &M, const GlobalAlias &GA) {
 
 void NVPTXAsmPrinter::emitHeader(Module &M, raw_ostream &O,
                                  const NVPTXSubtarget &STI) {
-  O << "//\n";
-  O << "// Generated by LLVM NVPTX Back-End\n";
-  O << "//\n";
-  O << "\n";
+  const unsigned PTXVersion = STI.getPTXVersion();
 
-  unsigned PTXVersion = STI.getPTXVersion();
-  O << ".version " << (PTXVersion / 10) << "." << (PTXVersion % 10) << "\n";
-
-  O << ".target ";
-  O << STI.getTargetName();
+  O << "//\n"
+       "// Generated by LLVM NVPTX Back-End\n"
+       "//\n"
+       "\n"
+    << ".version " << (PTXVersion / 10) << "." << (PTXVersion % 10) << "\n"
```

Comment:
We could probably make use of Format.h to simplify printing further:
https://github.com/llvm/llvm-project/blob/934c97dd16ae8048ae009d97be7d80db4e1dee66/llvm/include/llvm/Support/Format.h#L9-L18

Reply:
I think what we have now is maybe a little clearer and avoids converting the integer to a float.

Reply:
I didn't mean using floats -- that was just an example in the format.h. You can use other format specifiers like `%d`.
It uses `snprintf` under the hood.

Here's a better example:
https://github.com/llvm/llvm-project/blob/f719d8ac1ccd08a9e8def796ecb9b1cd517c86af/clang/tools/libclang/CIndex.cpp#L6626-L6631